app/extensions/restManager/RestManager.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=no-self-use
+"""
+Rest Manager base class, to be used for any entity (EDM/ACM) where we need to interface with an external
+system using REST API
+
+"""
+import logging
+from werkzeug.exceptions import BadRequest
+from flask import current_app, request, session, render_template  # NOQA
+from flask_login import current_user  # NOQA
+import requests
+from collections import namedtuple
+import utool as ut
+import json
+import keyword
+import uuid
+
+KEYWORD_SET = set(keyword.kwlist)
+
+log = logging.getLogger(__name__)
+
+
+def _json_object_hook(data):
+    keys = list(data.keys())
+    keys_set = set(keys)
+    keys_ = []
+    for key in keys:
+        if key in KEYWORD_SET:
+            key_ = '%s_' % (key,)
+            assert key_ not in keys_set
+        else:
+            key_ = key
+        keys_.append(key_)
+    values = data.values()
+    data_ = namedtuple('obj', keys_)(*values)
+    return data_
+
+
+# Not entirely certain how appropriate this is as a Mixin as it's dependent on a couple of things in
+# the main class
+class RestManagerUserMixin(object):
+    def check_user_login(self, username, password):
+        self._ensure_initialized()
+
+        success = False
+        for target in self.targets:
+            # Create temporary session
+            temporary_session = requests.Session()
+            try:
+                response = self._get(
+                    'session.login',
+                    username,
+                    password,
+                    target=target,
+                    target_session=temporary_session,
+                )
+
+                assert response.ok
+                decoded_response = response.json()
+                assert decoded_response['message']['key'] == 'success'
+
+                log.info(f'User authenticated via {self.NAME} (target = {target})')
+                success = True
+                break
+            except Exception:
+                pass
+            finally:
+                # Cleanup temporary session
+                temporary_session.cookies.clear_session_cookies()
+                temporary_session.close()
+
+        return success
+
+
+class RestManager(RestManagerUserMixin):
+    # pylint: disable=abstract-method
+
+    ENDPOINT_PREFIX = None
+    ENDPOINTS = None
+    NAME = None
+
+    def __init__(self, app, pre_initialize=False, *args, **kwargs):
+        super(RestManager, self).__init__(*args, **kwargs)
+        self.initialized = False
+
+        # Must be overwritten by the derived class
+        assert self.NAME is not None
+        assert self.ENDPOINT_PREFIX is not None
+        assert self.ENDPOINTS is not None
+
+        self.app = app
+        self.targets = set([])
+
+        app.edm = self
```

Comment:
Well spotted, that's a bug :)

Reply:
IMO this kind of attribute assignment to the app object should happen in the `init_app` space. I feel like it's a good idea for the app to only be used as reference and for registration during a manager object's init.

I'm also not a fan of `self.app = app` assignment. This kind of thing is frowned upon by in the Flask documentation, but lots of developers do it anyway. The better practice is to use the app as needed (or better yet a Blueprint), but not store it within the object. Instead use `flask.current_app` when you need to reference the current application in your object's logic.

Reply:
Those bits were just copied from the Existing EDMManager and tweaked but I see both your points. 
Fixed them. Will do a bit more testing today.